reporter/reporter.go

```diff
@@ -245,20 +252,21 @@ func (t *reporter) reportDerivedMetrics(span tracer.RawSpan) {
 	replaceTag(tags, "application", appName, appFound)
 	replaceTag(tags, "service", serviceName, svcFound)
 
-	if len(t.redMetricsCustomTagKeys) > 0 {
-		redMetricsCustomTags := make(map[string]string)
-		customTagMatch := false
-		for _, key := range t.redMetricsCustomTagKeys {
-			if value, found := getAppTag(key, "", span.Tags); found {
-				tags[key] = value
-				redMetricsCustomTags[key] = value
-				customTagMatch = true
-			}
-		}
-		if customTagMatch {
-			t.heartbeater.AddCustomTags(redMetricsCustomTags)
+	if _, ok := t.redMetricsCustomTagKeys[string(ext.SpanKind)]; !ok {
+		t.redMetricsCustomTagKeys[string(ext.SpanKind)] = exists
+	}
+	redMetricsCustomTags := tags
```

Comment:
This changes the previous behavior. Any particular reason you are setting this to tags?

Reply:
We might want to dynamically populate `span.kind` values for a given application and service from the heartbeat. It follows the change in Java SDK: https://wavefront.atlassian.net/browse/MONIT-18085